### ISSUE TITLE
Intellisense Prompt for Glossary does not work since v9.6.1

### ISSUE THREAD

**Issue by chigix** [James-Yu/LaTeX-Workshop#3851]
> _Please fill the following fields with a star (*) and provide as much related information as possible._

## Pre-checks*
> _Please change the following `[ ]` to `[x]` for confirmation._
- [x] The issue has not been reported in this repository.
- [x] The issue remains after disabling all other extensions and restarting Visual Studio Code.
- [x] The [FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) cannot address the issue.
- [x] The issue is not related to compiling a document, or the document can be successfully compiled in the OS terminal but not in Visual Studio Code with this extension.

## Environment*
> _Please write exact version numbers instead of descriptors such as `latest`._
- Operating System: Windows 10
- Visual Studio Code Version: 1.77.3
- LaTeX Workshop Version: 9.6.1~9.8.2
- TeX Distribution Version: TeX Live 2021

## The Issue*
> _Please briefly describe the issue you come across._
- Till LaTeX workshop v9.6.0, glossaries defined in `\newglossaryentry` can be prompted in intellisense suggest.
- Since LaTeX workshop v9.6.1, glossaries cannot be auto-suggested and in the latest version, only the following items appear in the suggestion candidates:
![image](https://user-images.githubusercontent.com/2692787/233639029-89393a17-b146-47ca-ae1c-1c90f4ae7fbc.png)

### Reproduction Steps
> _Please list out the steps to reproduce your bug. Include relevant environmental variables or any other configuration._
1. Install LaTeX workshop equal to or greater than v9.6.1
2. Insert a `\newglossaryentry` and define a term.
3. Input `\gls{` to trigger intellisense for glossaries.

### Expected Behavior
> _What were you expecting to see? Include any relevant examples or documentation links._
- Glossaries defined in `\newglossaryentry`s should appear in the intellisense suggested list.

**Comment by James-Yu** [comment]
I cannot reproduce the issue. We have a test named `glossary intellisense` to check glossay intellisense, which was passing ever since. You may need to provide a minimal working example to demonstrate the bug.

**Comment by neo-apz** [comment]
I'm having the exact same problem and I also see the exact same suggestions as the image above.

After a little bit of debugging I realized my issue seems to be that I have defined my own macro (wrapper) called `\newword` around `\newglossaryentry` and I define my terms using this wrapper, which in turn uses `\newglossaryentry` and does a few other things.

Is it possible to also identify the terms in this case and add them to the list of suggestions? As a general question, is it possible to have intellisense working with any custom macro and tell it how to figure out the suggestions?
